helper/validation.go

```diff
@@ -47,6 +47,16 @@ func (validators Validators) Check(obj map[string]any) error {
 	return nil
 }
 
+func GetValue(key string, v map[string]any) any {
+	keys := strings.Split(key, ".")
+	valueToCheck := v[keys[0]]
+	for i:= 1; i < len(keys); i++ {
+		valueToCheck = valueToCheck.([]interface{})[0].(map[string]any)[keys[i]]
+	}
+
+	return valueToCheck
```

Comment:
Is is normal to have no "if" statement in this for loop ? Wouldn't it always return the last key in the list ?

Reply:
That's how it's meant so if the key is `value.child.subchild` it will do `v["value"]["child"]["subchild"]`

Reply:
Ok but what is the purpose of this for loop? It looks like it does `valueToCheck = valueToCheck.([]interface{})[0].(map[string]any)[keys[len(keys)]]`

Reply:
it's to be able to move into the child element, with the previous example:
- line 52: we do `valueToCheck = v["value"]`
- line 54, the loop: we do `valueToCheck = valueToCheck["child"]` --> this is basically `v["value"]["child"]`
- line 54, step 2: we do `valueToCheck = valueToCheck["subchild"]` --> this is basically `v["value"]["child"]["subchild"]`

The loop is meant to go deeper into the object following the key "path"

Reply:
@elodietinland for a key like "a.b.c", it would do `v['a'][0]['b'][0]['c']`, because nested objects in Terraform are placed into arrays (there is no proper support for nested objects :/ )